### DISCUSSION TITLE
Sumatrapdf staying open after running command

### DISCUSSION THREAD

**Discussion by strjbyu** [sumatrapdfreader/sumatrapdf#4525] [Q&A]
Running the command as follows:

C:\Program Files\SumatraPDF\SumatraPDF.exe -print-to "printer" "file"

In task manager, I'm seeing lots and lots of instances of SumatraPDF.

How should I close/kill sumatraPDF after it runs and prints?

![image](https://github.com/user-attachments/assets/ec6980ed-46c6-44fd-ae96-3fc90447c9c8)

**Comment by GitHubRulesOK** [comment]
I dont think that is normal as not seen it before so it is perhaps your environment ?

Exact Version and ensure you explain how/who  is running it. That is NOT a valid command

clearly the nearest attempt to run with a fail is
`"C:\Program Files\SumatraPDF\SumatraPDF.exe" -print-to "printer" "out.pdf"`

"C:\Program Files\SumatraPDF\SumatraPDF.exe" -print-to "Microsoft Print to PDF" "out.pdf"

runs and closes for me
![image](https://github.com/user-attachments/assets/462b0b93-7ff3-4b2f-91c8-57015b737630)
![image](https://github.com/user-attachments/assets/985c14e1-f47b-4bd7-aba7-8e2798132bf8)

**Reply by strjbyu** [reply]
Where "printer" is listed in my command is the actual name of the printer for me. Also where the "file" is listed in the command, is the filepath and filename. 

The command prints just fine, but doesn't seem to reliably close it seems.

The version of SumatraPDF is 3.5.2, running on a Windows Server 2019. Running as administrator domain user on the system. It is being called from powershell,
param (
    [string]$file = "",
    [string]$printer = ""
)
echo $file
echo $printer
$arguments = "-print-to `"$printer`" `"$file`""
echo $arguments
Start-Process -FilePath "C:\Program Files\SumatraPDF\SumatraPDF.exe" -ArgumentList "$arguments"

which is then being called from php from within a function.
$command = 'powershell.exe C:\printPDF.ps1 -file "'.$filePath.'" -printer "'.$printerName.'"';
exec($command,$output,$returnVar);

**Reply by GitHubRulesOK** [reply]
Well thats why.

SumatraPDF is not designed for server usage it is entirely designed to be used by individuals direct from a command line.
It is the Same as Acrobat is not designed for serving even if Microsoft use it in their commercial server based printing!!
Acrobat would also be seen as unclosed as it needs to be re-used for multiple prints, but in their case staying resident is by design.

The best I can suggest  is use GhostScript which is perhaps the best designed for many such uses. It will be leaner and faster in most cases too. HOWEVER dont wrap printing in other languages it is designed like Chrome-headless-shell to be driven from CMD.EXE
